x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -121,13 +133,17 @@ boolean match(int stage, Iterable<Tuple<KeyAndOrdinal, HitReference>> hits) {
             }
         }
 
+        boolean matched;
         // check tail limit
         if (tailLimitReached()) {
             log.trace("(Tail) Limit reached {}", stats);
-            return false;
+            matched = false;
+        } else {
+            log.trace("{}", stats);
+            matched = true;
         }
-        log.trace("{}", stats);
-        return true;
+        trackMemory(ramBytesUsedInFlight, ramBytesUsedCompleted);
```

Comment:
I noticed that `trackMemory` is not called in the `if (headLimit) {...` branch. Why is that?

Reply:
Because it's an early exit, and no new objects are created/added in the structures.